### PR TITLE
files.js generator

### PR DESCRIPTION
**Pull request by mcarcaso**
A tool for generating files.js that'll be used with the build tool.

### CODE REVIEW

File: src/foam/build/FilesJsGen.js [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+    {
+      name: 'modelDAO',
+      expression: function(srcDir) {
+        return this.DirCrawlModelDAO.create({srcDir: srcDir})
+      },
+    },
+  ],
+  methods: [
+    function execute() {
+      this.getFilesJs().then(console.log);
```
Comment (adamvy) [inline src/foam/build/FilesJsGen.js]: I think you should console.log.bind(console), its not pre-bound on all systems.

File: src/foam/build/FilesJsGen.js [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+      if ( foam.Array.isInstance(o) ) {
+        return Promise.all(o.map(function(i) {
+          return self.getDepsTree(i, seen, head);
+        })).then(function() { return head });
+      } else if ( foam.Object.isInstance(o) ) {
+        // Check if it's an actual class. foam.core.FObject.isSubClass
+        // should work but doesn't:
+        // https://github.com/foam-framework/foam2/issues/1023
+        if ( o && o.prototype && (
+            foam.core.FObject.prototype === o.prototype ||
```
Comment (adamvy) [inline src/foam/build/FilesJsGen.js]: Indentation here is confusing, move the open parenthesis to this line so it read

```
if ( o && o.prototype &&
     ( foam.core.FObject... ||
       foam.core.FObject... ) ) {
```

Comment (mcarcaso) [reply src/foam/build/FilesJsGen.js]: Ah yeah that's nicer. Fixed here and in the JsCodeOutputter.

File: src/foam/build/FilesJsGen.js (hunk 1 above)
Comment (mcarcaso) [reply src/foam/build/FilesJsGen.js]: Done

**Comment by adamvy** [comment]
Merging as is because I haven't come up with a good TREE mlang yet.  But if I think of one I'll let you know.

**Pull request merged**